File: server/kafka-backend/src/core/domain/events/transfer-completed.event.ts
export class TransferCompletedEvent
{
  constructor(
    public readonly trackingId: string,
    public readonly fromAccountId: string,
    public readonly toAccountId: string,
    public readonly fromIban: string,
    public readonly toIban: string,
    public readonly amount: number,
    public readonly currency: string,
    public readonly completedAt: Date = new Date(),
  ) {}

  static eventName = 'transfer.completed';
}

File: server/java-backend/src/main/java/com/intbank/core/domain/event/TransferCompletedEvent.java
package com.intbank.core.domain.event;

import java.time.Instant;

public record TransferCompletedEvent(
    String trackingId,
    String fromAccountId,
    String toAccountId,
    String fromIban,
    String toIban,
    double amount,
    String currency,
    Instant completedAt
)
{
    public static final String EVENT_NAME = "transfer.completed";
}